src/auth/auth.service.ts

```diff
@@ -77,71 +72,81 @@ export class AuthService {
-      process.env.JWT_EXPIRY_TIME,
-    );
+    return accessTokenObject.accessToken;
   }
 
   async createPersonalAccessToken(userId: number): Promise<string> {
-    const validPersonalAccessToken =
-      await this.personalAccessTokenRepository.getValidPatForUserId(userId);
+    const payload = { userId };
+
+    const personalAccessTokenObject = await this.amqpConnection.request<{
+      personalAccessToken: string;
+    }>({
+      exchange: this.EXCHANGE,
+      routingKey: 'add-personal-access-token',
+      payload,
+    });
+    this.validateAuthMicroserviceReturn(personalAccessTokenObject);
 
-    if (validPersonalAccessToken) {
-      this.personalAccessTokenRepository.invalidatePatForUserId(userId);
-    }
-    const personalAccessToken = await this.generateToken(
-      userId,
-      process.env.JWT_PAT_SECRET,
-    );
-    const { token } =
-      await this.personalAccessTokenRepository.savePersonalAccessToken(
-        userId,
-        personalAccessToken,
-      );
-    return token;
+    return personalAccessTokenObject.personalAccessToken;
+  }
+
+  async validateAuthToken(token: string): Promise<number> {
+    const payload = { token };
+
+    const userIdObject = await this.amqpConnection.request<{ id: number }>({
+      exchange: this.EXCHANGE,
+      routingKey: 'validate-jwt-token',
+      payload,
+    });
+    this.validateAuthMicroserviceReturn(userIdObject);
+
+    return userIdObject.id;
   }
 
-  async validateUser(userRequest: UserRequest): Promise<UserPayloadRequest> {
+  async validateUser(userRequest: UserRequest): Promise<number> {
     const user = await this.userRepository.getUserByEmail(userRequest.email);
 
     if (!user) {
       throw new UnauthorizedException();
     }
 
-    const isMatch = await bcrypt.compare(userRequest.password, user.password);
+    const payload = { userId: user.id, password: userRequest.password };
 
-    if (!isMatch) {
-      throw new UnauthorizedException();
-    }
+    const validatedUserId = await this.amqpConnection.request<number>({
+      exchange: this.EXCHANGE,
+      routingKey: 'validate-user',
+      payload,
+    });
+    this.validateAuthMicroserviceReturn(validatedUserId);
 
-    return user;
+    return validatedUserId;
   }
 
-  async verifyAccountActivationToken(
-    jwtToken: string,
-  ): Promise<{ id: number }> {
-    const invalidTokenMessage =
-      'Invalid token. Please provide a valid token to activate account';
-
-    try {
-      return this.jwtService.verify(jwtToken, {
-        secret: process.env.JWT_ACCOUNT_ACTIVATION_SECRET,
-      });
-    } catch {
-      throw new UnauthorizedException(invalidTokenMessage);
-    }
-  }
+  async activateAccount(token: string): Promise<UserPayloadRequest> {
+    const payload = { token };
+
+    const userId = await this.amqpConnection.request<number>({
+      exchange: this.EXCHANGE,
+      routingKey: 'activate-account',
+      payload,
+    });
+    this.validateAuthMicroserviceReturn(userId);
```

Comment:
I like to put 1 empty line in such cases (after constructs that take more than 1 line)